src/main/scala/scalismo/io/ImageIO.scala

```diff
@@ -218,6 +218,54 @@ object ImageIO {
     }
   }
 
+  /**
+   * Read a 3D Scalar Image, and possibly convert it to the requested voxel type.
+   *
+   * This method is similar to the [[read3DScalarImage]] method, except that it will convert the image to the requested voxel type if
+   * the type in the file is different, whereas [[read3DScalarImage]] will throw an exception in that case.
+   *
+   * @param file  image file to be read
+   * @param resampleOblique  flag to resample oblique images. This is only required when reading Nifti files containing an oblique image. See documentation above [[ImageIO]].
+   * @param favourQform  flag to favour the qform Nifti header entry over the sform one (which is by default favoured). See documentation above [[ImageIO]].
+   * @tparam S Voxel type of the image
+   *
+   */
+  def read3DScalarImageAsType[S: Scalar: TypeTag: ClassTag](file: File, resampleOblique: Boolean = false, favourQform: Boolean = false): Try[DiscreteScalarImage[_3D, S]] = {
```

Comment:
Good idea. The only problem with this kind of "flag options" is that they can become tedious and unmaintainable over time, especially if there are also default values and/or if people rely on the order. But in principle, yes, and I'm not against it.